app/src/main/java/com/quarkworks/apartmentgroceries/service/NetworkRequest.java

```diff
@@ -32,24 +35,39 @@ public NetworkRequest(UrlTemplate template, Callback callback) {
     @Override
     protected String doInBackground(Void... params) {
 
+        String url = template.getUrl();
+        String method = template.getMethod();
+
+        // add parameters from template
+        StringBuilder urlStringBuilder = new StringBuilder(url);
+        urlStringBuilder.append("?");
+
```

Comment:
Check to see if this is a GET method first. Otherwise, add the request body like you were planning.